packages/api/README.md

```diff
@@ -12,10 +12,67 @@ For the full documentation of the API see: https://nasa.github.io/cumulus-api
 
 ### Running the API locally
 
+To run the API locally using Localstack for AWS services without Earthdata authentication required:
+
+```bash
   $ npm run serve
+```
+
+To run the API locally using Localstack for AWS services with Earthdata authentication required:
+
+```bash
+  EARTHDATA_CLIENT_ID=<your_client_id> \
+    EARTHDATA_CLIENT_PASSWORD=<your_password> \
+    npm run serve-oauth \
+    -- --username <your_username> # the extra `--` is required
+```
+
+**Note**: The Cumulus API checks whether the username used to login with Earthdata is an allowed user for the API. In order to add your Earthdata username as an allowed user when running the API against Localstack, you must specify it using the `--username` option.
+
+To run the API locally using your deployed stack with Earthdata authentication required:
+
+```bash
+  stackName=<your_stack_name> \
+    system_bucket=<your_system_bucket> \
+    EARTHDATA_CLIENT_ID=<your_client_id> \
+    EARTHDATA_CLIENT_PASSWORD=<your_password> \
+    ES_HOST=<your_elasticsearch_host> \
+    npm run serve-remote
+```
+
+If you want endpoints that interact with Elasticsearch to work, you must specify the `ES_HOST` environment variable for this command. You can get the Elasticsearch domain for your stack using the AWS CLI:
+
+```bash
+  aws es describe-elasticsearch-domain --domain-name <your_es_domain_name> --query 'DomainStatus.Endpoint'
+```
+
+### Running the distribution API locally
+
+To run the distribution API locally using Localstack for AWS services without Earthdata authentication required:
+
+```bash
+  $ npm run serve-dist
```

Comment:
I think the middlewares have changed this.  We might need a ticket to do something like this
https://github.com/nasa/cumulus/blob/master/packages/api/app/routes.js#L27-L30
because the distribution_api requires each req object to have `req.authorizedMetadata.userName` set  see: https://github.com/nasa/cumulus/blob/master/packages/api/endpoints/distribution.js#L137

Reply:
so I can run the dist endpoint, but can't do anything with it.
```
Starting server on port 5002
[26/Mar/2019:23:00:35 +0000] "GET /s3credentials HTTP/1.1"500 96 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.86 Safari/537.36" {"stack":"TypeError: Cannot read property 'userName' of undefined\n    at s3credentials (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/endpoints/s3credentials.js:41:43)\n    at handleCredentialRequest (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/endpoints/distribution.js:161:10)\n    at handleReturn (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:24:27)\n    at /Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:56:9\n    at handleReturn (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:24:27)\n    at /Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:56:9\n    at Layer.handle [as handle_request] (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express/lib/router/layer.js:95:5)\n    at next (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express/lib/router/route.js:137:13)\n    at ensureAuthorizedOrRedirect (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/endpoints/distribution.js:119:12)\n    at handleReturn (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:24:27)","message":"Cannot read property 'userName' of undefined"}
[26/Mar/2019:23:00:36 +0000] "GET /favicon.ico HTTP/1.1"404 100 "http://localhost:5002/s3credentials" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.86 Safari/537.36" -
[26/Mar/2019:23:01:01 +0000] "GET /s3credentials HTTP/1.1"500 96 "-" "-" {"stack":"TypeError: Cannot read property 'userName' of undefined\n    at s3credentials (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/endpoints/s3credentials.js:41:43)\n    at handleCredentialRequest (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/endpoints/distribution.js:161:10)\n    at handleReturn (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:24:27)\n    at /Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:56:9\n    at handleReturn (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:24:27)\n    at /Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:56:9\n    at Layer.handle [as handle_request] (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express/lib/router/layer.js:95:5)\n    at next (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express/lib/router/route.js:137:13)\n    at ensureAuthorizedOrRedirect (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/endpoints/distribution.js:119:12)\n    at handleReturn (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:24:27)","message":"Cannot read property 'userName' of undefined"}
[26/Mar/2019:23:03:32 +0000] "GET /localbucket/localrun/workflows/list.json HTTP/1.1"500 96 "-" "-" {"stack":"TypeError: Cannot read property 'userName' of undefined\n    at handleFileRequest (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/endpoints/distribution.js:190:28)\n    at handleReturn (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:24:27)\n    at /Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:56:9\n    at handleReturn (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:24:27)\n    at /Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:56:9\n    at Layer.handle [as handle_request] (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express/lib/router/layer.js:95:5)\n    at next (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express/lib/router/route.js:137:13)\n    at ensureAuthorizedOrRedirect (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/endpoints/distribution.js:119:12)\n    at handleReturn (/Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:24:27)\n    at /Users/savoie/projects/cumulus/898-use-local-API/packages/api/node_modules/express-promise-router/lib/express-promise-router.js:56:9","message":"Cannot read property 'userName' of undefined"}
```

Reply:
I re-factored a bit to mimic what the regular API router does for faking authentication, but in the case of the distribution API, it doesn't seem necessary to create an entirely separate middleware for fake auth. Also, I tested the above endpoints and they should work now, except that `/s3credentials` can't find the lambda from local runs, but you would still be able to debug it